common/fs/src/cache/dir_path.rs

```diff
@@ -74,3 +75,59 @@ impl From<DirPathBuf> for PathBuf {
         d.inner
     }
 }
+
+// Recursive directory back off
+fn find_valid_path(
+    path: Option<PathBuf>,
+    postfix: Option<PathBuf>,
+) -> Result<DirPathBuf, DirPathBufError> {
+    match path {
+        Some(p) => {
+            if p.is_dir() {
+                Ok(DirPathBuf { inner: p, postfix })
+            } else {
+                warn!("{:?} is not a directory; moving to parent directory", p);
+                let mut postfix_pathbuf = PathBuf::new();
+                let mut pop_pathbuf = PathBuf::new();
+                if let Some(path) = postfix {
```

Comment:
shadowing fn arg on purpose?

Reply:
The `postfix` function input is only non-None in the event that a given directory doesn't exist, in which case it saves that piece of the path so it can properly construct the missing directory path.

Reply:
i was thinking about var name

Reply:
Ah, I see. Updated var name `path` -> `pop`